### DISCUSSION TITLE
choco command takes time to initialise. How to skip `RemovePendingPackagesTask`?

### DISCUSSION THREAD

**Discussion by jimis** [chocolatey/choco#2186] [Q&A]
On the Windows runners in Github Actions, there are a bunch of Packages installed already. Because of that it takes a long time for the first run of `choco` command.

As evident from the debug log it spends around 2 minutes in `RemovePendingPackagesTask`:

```
Tue, 19 Jan 2021 11:21:36 GMT Received arguments: install ccache --version 3.7.12 --yes --no-progress --not-silent --verbose --debug
Tue, 19 Jan 2021 11:21:36 GMT RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
Tue, 19 Jan 2021 11:21:36 GMT Sending message 'PreRunMessage' out if there are subscribers...
Tue, 19 Jan 2021 11:21:36 GMT [Pending] Removing all pending packages that should not be considered installed...
Tue, 19 Jan 2021 11:23:17 GMT Performing validation checks.
```

Is there an option to skip this step?

**Comment by gep13** [comment]
Is this repeatable? Do you have an example workflow that you are using?

**Reply by jimis** [reply]
https://github.com/qt/qtbase/runs/1754601585
(EDIT: workflow listed [here](https://github.com/qt/qtbase/actions/runs/506067778/workflow))

From the "..." button enable the display of timestamps, then check the "Prepare Windows" step.

I've been seeing this since a long time ago even though I hadn't traced it, so I'd say it's repeatable.

**Reply by jimis** [reply]
Could issue https://github.com/chocolatey/choco/issues/1866 be relevant? There is also pull request https://github.com/chocolatey/choco/pull/1941 that supposedly resolves it.

**Reply by jetersen** [reply]
I created a repro here: https://github.com/jetersen/test-choco-pack-github-action
See https://github.com/actions/virtual-environments/issues/3010
